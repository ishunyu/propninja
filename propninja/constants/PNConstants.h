//
//  PNConstants.h
//  propninja
//
//  Created by Shun Yu on 12/24/14.
//  Copyright (c) 2014 Workday. All rights reserved.
//

#import <Foundation/Foundation.h>

extern NSString *ERROR;
extern long ERROR_INPUTSTREAM;
extern long ERROR_OUTPUTSTREAM;

extern NSString *PATH_SCRIPTS;
extern NSString *PATH_SERVER_LOG;
extern NSString *PATH_PYTHON_INDEXER;
extern NSString *PATH_PYTHON_PROPERTIES;
extern NSString *PATH_PYTHON_SERVER;
extern NSString *PATH_PYTHON_WHOOSH;
extern NSString *PATH_STANDARD_INPUT_OUTPUT_PROPERTIES_SERVER;

extern NSString *FILE_TYPE_PYTHON;
extern NSString *FILE_TYPE_LOCK;
extern NSString *FILE_TYPE_JSON;

extern NSString *KEY_CONFIG_HOTKEY;
extern NSUInteger VALUE_CONFIG_HOTKEY_DEFAULT_MODIFIER_FLAGS;
extern unsigned char VALUE_CONFIG_HOTKEY_DEFAULT_KEY_CODE;

extern NSString *KEY_PROPERTY_MODIFIER_FLAGS;
extern NSString *KEY_PROPERTY_KEY_CODE;

extern NSString *KEY_CONFIG_PROPERTY_FILES;
extern NSString *KEY_CONFIG_PROPERTY_USAGE_METRICS;
extern NSString *KEY_PROPERTY_USAGE_METRICS_DATA;
extern NSString *KEY_PROPERTY_USAGE_METRICS_ITEM_FILE;
extern NSString *KEY_PROPERTY_USAGE_METRICS_ITEM_KEY;

extern NSString *KEY_PROPERTY_TAG;
extern NSString *KEY_PROPERTY_PATH;
extern NSString *KEY_PROPERTY_ENABLED;
