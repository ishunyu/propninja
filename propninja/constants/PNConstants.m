//
//  PNConstants.m
//  propninja
//
//  Created by Shun Yu on 12/24/14.
//  Copyright (c) 2014 Workday. All rights reserved.
//

#import "PNConstants.h"

NSString *ERROR = @"PN_ERROR";
long ERROR_INPUTSTREAM = 1;
long ERROR_OUTPUTSTREAM = 2;

NSString *DIR_SCRIPTS = @"scripts";

NSString *FILE_TYPE_PYTHON = @"py";
NSString *FILE_TYPE_LOCK = @"lock";
NSString *FILE_TYPE_JSON = @"json";


NSString *FILE_PROPERTIES_SERVER = @"server";

NSString *KEY_CONFIG_HOTKEY = @"CONFIG_HOTKEY";
NSUInteger VALUE_CONFIG_HOTKEY_DEFAULT_MODIFIER_FLAGS = NSAlternateKeyMask | NSCommandKeyMask;
unsigned char VALUE_CONFIG_HOTKEY_DEFAULT_KEY_CODE = 47;

NSString *KEY_PROPERTY_MODIFIER_FLAGS = @"modifierFlags";
NSString *KEY_PROPERTY_KEY_CODE = @"keyCode";

NSString *KEY_CONFIG_PROPERTY_FILES = @"CONFIG_PROPERTY_FILES";

NSString *KEY_PROPERTY_TAG = @"tag";
NSString *KEY_PROPERTY_PATH = @"path";
NSString *KEY_PROPERTY_ENABLED = @"enabled";


