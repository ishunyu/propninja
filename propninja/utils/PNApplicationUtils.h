//
//  PNApplicationUtils.h
//  propninja
//
//  Created by Shun Yu on 8/14/16.
//  Copyright Â© 2016 Workday. All rights reserved.
//

#import <Foundation/Foundation.h>

@interface PNApplicationUtils : NSObject
+ (void)terminate;
@end
