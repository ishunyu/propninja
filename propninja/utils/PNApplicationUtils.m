//
//  PNApplicationUtils.m
//  propninja
//
//  Created by Shun Yu on 8/14/16.
//  Copyright Â© 2016 Workday. All rights reserved.
//

#import "PNApplicationUtils.h"

@implementation PNApplicationUtils
+ (void)terminate
{
    [[NSApplication sharedApplication] terminate:nil];
}
@end
