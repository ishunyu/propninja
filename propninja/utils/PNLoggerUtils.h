//
//  PNLoggerUtils.h
//  propninja
//
//  Created by Shun Yu on 1/31/15.
//  Copyright (c) 2015 Workday. All rights reserved.
//

#import <Foundation/Foundation.h>

#import <CocoaLumberjack/DDLog.h>
#import <CocoaLumberjack/DDTTYLogger.h>
#import <CocoaLumberjack/DDFileLogger.h>

//static const int ddLogLevel = LOG_LEVEL_INFO;
static const int ddLogLevel = LOG_LEVEL_DEBUG;
//static const int ddLogLevel = LOG_LEVEL_VERBOSE;

@interface PNLoggerUtils : NSObject

@end

@interface PNLineNumberLogFormatter : NSObject<DDLogFormatter>

- (NSString *)formatLogMessage:(DDLogMessage *)logMessage;

@end
