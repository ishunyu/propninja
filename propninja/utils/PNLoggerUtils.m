//
//  PNLoggerUtils.m
//  propninja
//
//  Created by Shun Yu on 1/31/15.
//  Copyright (c) 2015 Workday. All rights reserved.
//

#import "PNLoggerUtils.h"

@implementation PNLoggerUtils

+ (void)load
{
    [[DDTTYLogger sharedInstance] setLogFormatter:[[PNLineNumberLogFormatter alloc] init]];
    [DDLog addLogger:[DDTTYLogger sharedInstance] withLogLevel:LOG_LEVEL_ALL];
    
    DDFileLogger *fileLogger = [[DDFileLogger alloc] init];
    fileLogger.rollingFrequency = 60 * 60 * 24; // 24 hour rolling
    fileLogger.logFileManager.maximumNumberOfLogFiles = 7;
    [fileLogger setLogFormatter:[[PNLineNumberLogFormatter alloc] init]];
    
    [DDLog addLogger:fileLogger withLogLevel:LOG_LEVEL_INFO];
}

@end

@interface PNLineNumberLogFormatter ()
@property (strong, nonatomic) NSDateFormatter *formatter;
@end

@implementation PNLineNumberLogFormatter
- (id)init
{
    if (self = [super init]) {
        self.formatter = [[NSDateFormatter alloc] init];
        [self.formatter setFormatterBehavior:NSDateFormatterBehavior10_4];
        [self.formatter setDateFormat:@"yyyy-MM-dd HH:mm:ss:SSS"];
    }
    
    return self;
}

- (NSString *)formatLogMessage:(DDLogMessage *)logMessage
{
    NSString *timestamp = [self.formatter stringFromDate:logMessage->timestamp];
    NSString *logLevel = [self logLevel:logMessage->logFlag];
    NSString *path = [NSString stringWithCString:logMessage->file encoding:NSASCIIStringEncoding];
    NSString *fileLineNumber = [NSString stringWithFormat:@"%@:%d", [path lastPathComponent], logMessage->lineNumber];
    
    return [NSString stringWithFormat:@"%@ [%@] %-32s %@",timestamp, logLevel, [fileLineNumber UTF8String], logMessage->logMsg];
}

- (NSString *)logLevel:(int)logFlag
{
    switch (logFlag)
    {
        case LOG_FLAG_ERROR :
            return @"E";
        case LOG_FLAG_WARN  :
            return @"W";
        case LOG_FLAG_INFO  :
            return @"I";
        case LOG_FLAG_DEBUG :
            return @"D";
        default             :
            return @"V";
    }
}
@end