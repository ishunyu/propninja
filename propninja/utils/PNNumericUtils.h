//
//  PNNumericUtils.h
//  propninja
//
//  Created by Shun Yu on 1/30/15.
//  Copyright (c) 2015 Workday. All rights reserved.
//

#import <Foundation/Foundation.h>

@interface PNNumericUtils : NSObject
+ (int)randomNumberBetween:(int)min max:(int)max;
@end
