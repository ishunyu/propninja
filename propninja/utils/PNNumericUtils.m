//
//  PNNumericUtils.m
//  propninja
//
//  Created by Shun Yu on 1/30/15.
//  Copyright (c) 2015 Workday. All rights reserved.
//

#import "PNNumericUtils.h"

@implementation PNNumericUtils

+ (int)randomNumberBetween:(int)min max:(int)max
{
    return min + arc4random_uniform(max - min + 1);
}

@end
