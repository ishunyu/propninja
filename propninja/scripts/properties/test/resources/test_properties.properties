####################################
# test_properties.properties
# Used for testing Properties
####################################

# Whitespaces after key
keyA1
keyA1.1 
keyA1.2  
keyA1.3	

# Whitespaces as separator
keyA2 value
keyA2.1  value

# Separator & whitespaces
keyA3:value
keyA3.1: value
keyA3.2 : value
keyA4=value
keyA4.1= value
keyA4.2 = value

# Special character handling
keyA5=value#
keyA5.1=value!
keyA6=\v\a\lu\e
keyA7=v a l u e
keyA8==value

# There is a space after 'value'
keyA9=value 
# There is a tab after 'value'
keyA9.1=value	

# More comments
! Even more comments
 # Comments with space before it
 ! Comments with space before it
  # Comments with spaces before it
	# Comments with tab before it
		# Comments with tabs before it
	 	 	 	# Comments with tabs and spaces before it

# Spaces below
        
			

# Multiline properties
keyB=valu\
e
keyB1=val\
ue
keyB2=v\
a\
l\
u\
e
keyB2.1= v\
	a\
 l\
u\
	 	e
keyB3=value\

keyB3.1=value\
 
keyB4=value\\\

# Special character handling (key)
\k\e\y\C\1=value
 keyC2=value
\ keyC3=value
\	keyC4=value
\\keyC5=value

# Comments before last line
