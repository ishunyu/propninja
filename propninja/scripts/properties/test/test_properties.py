#!/usr/bin/env python

"""
Run under project root
    python -m unittest test.test_properties

Tests correctness of the properties injested by Properties class
    - Injest .properties files from TEST_RESOURCE_DIR
    - Injest respective .expected files
    - Compare the values with expected
    - Write out Properties to TEST_TMP_DIR
    - Compare the output file with the original
"""

import os
import random
import filecmp
import unittest

from properties.properties import Properties
from constants import *
from utils import *

class TestProperties(unittest.TestCase):
    def setUp(self):
        print "cwd:", os.getcwd()
        print "TEST_RESOURCE_DIR:", TEST_RESOURCE_DIR
        print "isdir:", os.path.isdir(TEST_RESOURCE_DIR)
        _, _, files = next(os.walk(TEST_RESOURCE_DIR))
        self.properties_files = [f for f in files if f.endswith(".properties")]

    def test_properties(self):
        for f in self.properties_files:
            fpath = os.path.join(TEST_RESOURCE_DIR, f)
            props = read(fpath)
            expected = readExpected(fpath + ".expected")
            
            self.assertEqual(len(expected), len(props))

            for k, v in expected.iteritems():
                self.assertEqual(v, props[k], "k: '%s', v: '%s', props[k]: '%s'" % (k, v, props[k]))

            pfpath = os.path.join(TEST_TMP_DIR, f)
            write(pfpath, props)

            is_same(fpath, pfpath)