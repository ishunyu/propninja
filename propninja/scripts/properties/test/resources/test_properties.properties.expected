keyA1

keyA1.1

keyA1.2

keyA1.3

keyA2
value
keyA2.1
value
keyA3
value
keyA3.1
value
keyA3.2
value
keyA4
value
keyA4.1
value
keyA4.2
value
keyA5
value#
keyA5.1
value!
keyA6
value
keyA7
v a l u e
keyA8
=value
keyA9
value 
keyA9.1
value	
keyB
value
keyB1
value
keyB2
value
keyB2.1
value
keyB3
value
keyB3.1
value
keyB4
value\
keyC1
value
keyC2
value
 keyC3
value
	keyC4
value
\keyC5
value